htlcswitch/test_utils.go

```diff
@@ -376,8 +376,9 @@ func getChanID(msg lnwire.Message) (lnwire.ChannelID, error) {
 
 // generatePayment generates the htlc add request by given path blob and
 // invoice which should be added by destination peer.
-func generatePayment(invoiceAmt, htlcAmt lnwire.MilliSatoshi, timelock uint32,
-	blob [lnwire.OnionPacketSize]byte) (*channeldb.Invoice, *lnwire.UpdateAddHTLC, error) {
+func generatePayment(invoiceAmt, htlcAmt lnwire.MilliSatoshi, htlcID uint64,
```

Comment:
Why's this necessary? During normal tests, this goes through the switch. Therefore, the lin that accepts the HTLC will have properly set the ID of the incoming HTLC.

Reply:
It is done because in the tests using the `singleLinkTestHarness` I am not using the switch for the remote side of the channel, making it necessary to set this ID manually. This is necassary only one single place in the test I think, so I can revert this change, and set it manually there.